rocAL/rocAL/source/api/rocal_api_data_transfer.cpp

```diff
@@ -57,8 +56,7 @@ rocalToTensor16(RocalContext p_context, half *out_ptr, RocalTensorLayout tensor_
     auto context = static_cast<Context*>(p_context);
     try
     {
-        auto tensor_layout = (tensor_format == ROCAL_NHWC) ?  RocalTensorFormat::NHWC : RocalTensorFormat::NCHW;
-        //auto tensor_output_data_type = (tensor_data_type == ROCAL_FP32) ? RocalTensorDataType::FP32 : RocalTensorDataType::FP16;
+        auto tensor_layout = (tensor_format == ROCAL_NHWC) ?  RocalTensorlayout::NHWC : RocalTensorlayout::NCHW;
         context->master_graph->to_tensor(out_ptr, tensor_layout, multiplier0, multiplier1, multiplier2,
                 offset0, offset1, offset2, reverse_channels, RocalTensorDataType::FP16, output_mem_type);
```

Comment:
This API specifically states rocalToTensor16, which means it will convert to a F16 tensor and we have one API for F32 as well
So changes can't be made to this API.

Reply:
Okay
Currently our use cases are only FP16 when we meant rocalToTensor16, but if we decide to make the api generic for all datatypes of 16 bit, i thought we can make the changes as suggested
Please think along these lines and make a call

Reply:
There is an API called copy_data available in the tensor.cpp which can copy to any data type
These ToTensor16 and ToTensor32 involve copy and normalization, So not sure how I16 can be used for these API
If you feel it is necessary to be included here we can discuss with @shobana-mcw